intercompany_shared_contact/models/res_partner.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Akretion (https://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    res_company_id = fields.One2many(
+        "res.company",
+        "partner_id",
+        readonly=True,
+        help="Effectively a One2one field to represent the corresponding res.company",
+    )
+    origin_company_id = fields.Many2one(
+        "res.company",
+        compute="_compute_origin_company_id",
+        store=True,
+        help="Hack field to keep the information of the 'real' company_id. "
+        "That way, we can share the contact by setting company_id to null, "
+        "without losing any information. If null, the contact is not shared.",
+    )
+
+    @api.depends("res_company_id", "parent_id.origin_company_id")
+    def _compute_origin_company_id(self):
+        for record in self:
+            if record.parent_id.origin_company_id:
+                record.origin_company_id = record.parent_id.origin_company_id
+                record.company_id = False
+            if record.res_company_id:
+                record.origin_company_id = record.res_company_id
+                record.company_id = False
+            else:
+                record.origin_company_id = False
+
+    # super().sudo().create() has some interaction that loops infinitely,
+    # thus the switch with multiple sudo() calls
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        if self.env.context.get("creating_res_company"):
```

Comment:
@kevinkhao This is a huge security issue.
Every time you use sudo becarefull.
With this code anybody without behind logged can create / update a partner.
So basically you can update the email of admin user, use forget password and you are the admin of the database !

Be really carefull !

Reply:
I am working on a fix